File: ehk-model/model/tweet.go
package model

// TweetRecord represents a tweet with user ID, step, and opinion
type TweetRecord struct {
	UserID  int
	Step    int
	Opinion float64
}

// Tweet represents a tweet object
type Tweet struct {
	User    int
	Step    int
	Opinion float64
}

// ToRecord converts a Tweet to a TweetRecord
func (t *Tweet) ToRecord() TweetRecord {
	return TweetRecord{
		UserID:  t.User,
		Step:    t.Step,
		Opinion: t.Opinion,
	}
}

File: ehk_model_old/base/tweet.py
from typing import Tuple, TypeAlias

import dataclasses

TweetRecord: TypeAlias = Tuple[int, int, float]

@dataclasses.dataclass
class Tweet:
  user: int
  step: int
  opinion: float
  
  def to_record(self) -> TweetRecord:
    return (self.user, self.step, self.opinion)
  